python/Ganga/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -52,8 +52,8 @@ def makeGangaListByRef(_list, preparable=False):
     result = GangaList()
     if len(_list) == 0:
         return result
-    temp_list = [stripProxy(element) for element in _list]
-    result._list = temp_list
+
+    result._list.extend([stripProxy(element) for element in _list])
```

Comment:
I'm happy to keep the changes to ListByRef, infact I support it, but we do need to intercept bad assignments to list as I've seen it abused in too many other places in the codebase in the past.

This may not be the case due to recent changes/fixes/developments but we need to prevent for people doing bad things to ```_list``` such as breaking ownership.

I'm weary that by changing this here we effect:
```
./Ganga/Core/GangaRepository/VStreamer.py
./Ganga/GPIDev/Lib/Job/Job.py
./GangaLHCb/Lib/LHCbDataset/LHCbDataset.py
```

We should check these files to be 1,000% sure that they should be copying by ref vs using this incorrectly and causing things to break.
If we verify all uses of this function are correct we can at least fix this small problem.